### PR TITLE
Adds actionbars to welding/unwelding the singulo emitters and field generators

### PR DESCRIPTION
**Pull request by TTerc**
[qol]

## About the PR 

This PR adds actionbars to welding/unwelding the singulo emitters and field generators.

## Why's this needed? 

New players often overlook the welding status chatbar messages, which, as a result, leads to a lot of unintentional singulooses.
With actionbars, it's more clear what you're supposed to do when welding the emitters/field gens to the floor.

## Changelog

```changelog
(u)TTerc
(+)Added actionbars to welding/unwelding singularity emitters and field generators.
```

### CODE REVIEW

File: code/obj/machinery/singularity.dm [hunk 1]
```diff
@@ -639,43 +639,19 @@ for some reason I brought it back and tried to clean it up a bit and I regret ev
 			return
 
 	if(isweldingtool(W))
-
-		var/turf/T = user.loc
-
+		var/datum/action/bar/icon/callback/action_bar = new /datum/action/bar/icon/callback(user, src, 2 SECONDS, /obj/machinery/field_generator/proc/weld_action,\
```
Comment (ZeWaka) [inline code/obj/machinery/singularity.dm:642]: why not use the generic one

File: code/obj/machinery/singularity.dm [hunk 2]
```diff
@@ -1118,6 +1080,19 @@ for some reason I brought it back and tried to clean it up a bit and I regret ev
 			if(M == user)	continue
 			M.show_message("<span class='alert'>The [src.name] has been hit with the [W.name] by [user.name]!</span>")
 
+/obj/machinery/emitter/proc/weld_action(mob/user)
+	if(state == 1)
+		state = 3
```
Comment (ZeWaka) [inline code/obj/machinery/singularity.dm:1085]: please `#define` these

File: code/obj/machinery/singularity.dm [hunk 3]
```diff
@@ -694,6 +670,18 @@ for some reason I brought it back and tried to clean it up a bit and I regret ev
 			if(M == user)	continue
 			M.show_message("<span class='alert'>The [src.name] has been hit with the [W.name] by [user.name]!</span>")
 
+/obj/machinery/field_generator/proc/weld_action(mob/user)
+	if(state == 1)
+		state = 3
```
Comment (ZeWaka) [inline code/obj/machinery/singularity.dm:675]: ditto

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/machinery/singularity.dm (hunk 1 above)
Comment (TTerc) [reply code/obj/machinery/singularity.dm:642]: I'm using `actions.start` twice per actionbar definition. I figured the code is cleaner that way

Comment (ZeWaka) [reply code/obj/machinery/singularity.dm:642]: @Adharainspace variant where it doesn't auto start, and returns the ref?

Comment (TTerc) [reply code/obj/machinery/singularity.dm]: okay, I actually managed to make it work with single generic actionbar per emitter/field gen

**Review by flrsh — APPROVED** [review]

**Pull request merged**